src/lib/styles/index.js

```diff
@@ -12,16 +12,28 @@ function isFunction(functionToCheck) {
  * @param {*} withStyleSheet wheather should or shouldn't be the result wrapped with `StyleSheet.create`
  */
 export const withStyles = (mapThemeToStyles, withStyleSheet = true) => Component => {
+  const getUpdatedStyles = props => {
+    let styles = {}
+    if (isFunction(mapThemeToStyles)) {
+      const stylesObject = mapThemeToStyles(props)
+      styles = withStyleSheet ? StyleSheet.create(stylesObject) : stylesObject
+    }
+    return styles
+  }
+
   class WrappedComponent extends React.Component {
     constructor(props) {
       super(props)
-      this.props = props
-      if (isFunction(mapThemeToStyles)) {
-        const stylesObject = mapThemeToStyles(this.props)
-        this.styles = withStyleSheet ? StyleSheet.create(stylesObject) : stylesObject
-      } else {
-        this.styles = {}
-      }
+    }
+
+    componentDidMount() {
+      const styles = getUpdatedStyles(this.props)
```

Comment:
This doesn't seem right.
This is going to cause a rerender all over.
Why was it necessary to move it away from the constructor?

Reply:
@sirpy having this in the constructor was causing that when updating the component props the styles were not updated using these new values.
Example: Changing color for a title in the modals. This was visible in error dialogs being show with darkGray titles
I did the change this way expecting only update the state when props are updated and not in all re-renders.